src/parser/mod.rs

```diff
@@ -5335,7 +5335,7 @@ impl<'a> Parser<'a> {
                 for_query: None,
             });
 
-            if self.next_token() != Token::Comma {
+            if self.peek_token().token == Token::SemiColon || self.next_token() != Token::Comma {
```

Comment:
Oh could this be simplified if we reused the `self.parse_comma_separated()` variant? I think if thats' possible it would be more conventional to use in the codebase (and would've prevented the edge case)

Reply:
@wugeer (just double checking since there was a requested review on this MR), did you get a chance to visit this comment yet?

just in case, to clarify I was thinking as the fix we could just move the body of the loop (minus line) into a method and call that method via parse_comma_separated. for example
```rust
    fn parse_mssql_declare_stmt(&mut self) -> Result<Declare, ParserError> {
        let stmts = self.parse_comma_separated(Parser::parse_mssql_declare_stmt)?;
        Ok(Statement::Declare { stmts })
    }
    fn parse_mssql_declare_stmt(&mut self) -> Result<Declare, ParserError> {
        let name = { ...};
        // ... remaining body verbatim
        let assignment = self.parse_mssql_variable_declaration_expression()?;
        Ok(Declare { names, ... })
    }
```